src/pudl/transform/ferc1.py

```diff
@@ -3866,3 +3885,474 @@ def plants_steam_ferc1(
+                #     xbrl_factoid: remove_nones_in_list(calc_components)
+                #     for (xbrl_factoid, calc_components) in meta_converted[
+                #         table_name
+                #     ].items()
+                # }
+        return meta_converted
+
+
+class TableCalcs:
+    """Methods to convert and check table-level calculations."""
+
+    def __init__(
+        self,
+        table_name: TableIdFerc1,
+        xbrl_meta_tbl: pd.DataFrame,
+        params: Ferc1TableTransformParams,
+    ):
+        """Instance of :class:`TableCalcs`."""
+        self.table_name = table_name
+        self.xbrl_meta_tbl = xbrl_meta_tbl
+        self.params = params
+
+    def rename_calcuations_xbrl_meta(
+        self,
+    ) -> dict[str, list]:
+        """Rename the calculations in the xbrl metadata to reflect PUDL names.
+
+        Note: params.select_dbf_rows_by_category.additional_categories will
+        include dbf categories that could be relevant.
+        """
+        xbrl_meta_tbl = self.xbrl_meta_tbl
+        calced_values = set(
+            xbrl_meta_tbl.loc[
+                xbrl_meta_tbl.row_type_xbrl == "calculated_value", "xbrl_factoid"
+            ]
+        )
+        calcs = {
+            calced_value: literal_eval(
+                xbrl_meta_tbl.set_index(["xbrl_factoid"]).loc[
+                    calced_value, "calculations"
+                ]
+            )
+            for calced_value in calced_values
+        }
+        calcs_renamed = {
+            self.rename_column_xbrl_to_pudl(calced_value): {
+                "calcs": [
+                    {
+                        key: self.rename_column_xbrl_to_pudl(value)
+                        if key == "name"
+                        else value
+                        for (key, value) in calc.items()
+                    }
+                    for calc in calcs
+                ],
+                "name_original": calced_value,
+            }
+            for (calced_value, calcs) in calcs.items()
+        }
+        names_non_calcs = [
+            name for name in xbrl_meta_tbl.xbrl_factoid if name not in calced_values
+        ]
+        non_calcs = {
+            self.rename_column_xbrl_to_pudl(name): {"name_original": name}
+            for name in names_non_calcs
+        }
+        return calcs_renamed | non_calcs
+
+    def rename_column_xbrl_to_pudl(
+        self,
+        col_name_xbrl: str,
+    ) -> str:
+        """Rename a column name from orignal XBRL name to the transformed PUDL name.
+
+        There are several transform params that either explicitly or implicity rename
+        columns:
+        * :class:`RenameColumnsFerc1`
+        * :class:`WideToTidySourceFerc1`
+        * :class:`UnstackBalancesToReportYearInstantXbrl`
+        * :class:`ConvertUnits`
+
+        This method attempts to use the table params to translate a column name.
+
+        Note: Instead of doing this for each individual column name, we could compile a
```

Comment:
This seems like it will be a generally useful piece of functionality, and I am pleasantly surprised that it seems to be working across the board so far! Given that this behavior is determined by several subcomponents of the `Ferc1TableTransformParams` and I can imagine that there might be new transforms & params which get added to that class in the future that have column renaming effects, I think it makes sense to have this be a method of `Ferc1TableTransformParams`. We could have a couple of methods that work together and are potentially useful on their own in different contexts, one to generate a rename mapping, one to rename elements in a dictionary, one to rename columns in a dataframe, etc.  The responsibility for remaining aware of how the various parameters and their methods result in columns getting renamed would then clearly live inside the class(es) actually doing the renaming rather than over here in metadataland.